packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1686,13 +1686,11 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
           bottomSpacing,
         );
       }
-      final Rect inflatedRect = Rect.fromLTRB(
-          newCaretRect.left - widget.scrollPadding.left,
-          newCaretRect.top - widget.scrollPadding.top,
-          newCaretRect.right + widget.scrollPadding.right,
-          newCaretRect.bottom + bottomSpacing,
-      );
-      _editableKey.currentContext.findRenderObject().showOnScreen(
+
+      final Rect inflatedRect = widget.scrollPadding
```

Comment:
An alternative to the trimming in `RenderSliverPersistentHeader.showOnScreen` is to confine `inflatedRect` so that it doesn't exceed `renderEditable`'s paintBounds. But that breaks around 6 tests.